static/app/components/workflowEngine/layout/breadcrumbs.tsx

```diff
@@ -0,0 +1,45 @@
+import {createContext, useContext} from 'react';
+
+import type {Crumb, CrumbDropdown} from 'sentry/components/breadcrumbs';
+import {Breadcrumbs} from 'sentry/components/breadcrumbs';
+import {useDocumentTitle} from 'sentry/components/sentryDocumentTitle';
+
+export interface BreadcrumbsContextValue {
+  crumbs: (Crumb | CrumbDropdown)[];
+}
+export const BreadcrumbsContext = createContext<BreadcrumbsContextValue>({
+  crumbs: [],
+});
+
+export function BreadcrumbsProvider({
+  children,
+  crumb,
+}: {
+  children: React.ReactNode;
+  crumb: Crumb | CrumbDropdown;
+}) {
+  const context = useContext(BreadcrumbsContext);
+  return (
+    <BreadcrumbsContext.Provider value={{crumbs: [...context.crumbs, crumb]}}>
+      {children}
+    </BreadcrumbsContext.Provider>
+  );
+}
+
+/**
+ * Automatically render Breadcrumbs from the <BreadcrumbsProvider /> and <SentryDocumentTitle /> contexts
+ */
+export function BreadcrumbsFromContext() {
+  const {crumbs = []} = useContext(BreadcrumbsContext);
+  const documentTitle = useDocumentTitle();
+  if (crumbs.length === 0) {
+    throw new Error(
+      `<BreadcrumbsFromContext> was not rendered inside of <BreadcrumbsProvider>!`
+    );
+  }
```

Comment:
I've seen this pattern used a few times to throw errors for contexts in our codebase, would we want to do the same here for `BreadcrumbsContext`?

https://github.com/getsentry/sentry/blob/9223abd010e31ea56693a9bc559c1d9115fff126/static/app/views/profiling/continuousProfileProvider.tsx#L45-L51